lib/spack/spack/util/editor.py

```diff
@@ -65,27 +76,30 @@ def editor(*args, **kwargs):
     searching the full list above, we'll raise an error.
 
     Arguments:
-        args (list): args to pass to editor
+        args: args to pass to editor
 
     Optional Arguments:
-        _exec_func (function): invoke this function instead of ``os.execv()``
-
+        exec_fn: invoke this function to run; use ``spack.util.editor.executable`` if you
+            want something that returns, instead of the default ``os.execv()``.
     """
-    # allow this to be customized for testing
-    _exec_func = kwargs.get("_exec_func", os.execv)
 
     def try_exec(exe, args, var=None):
         """Try to execute an editor with execv, and warn if it fails.
 
         Returns: (bool) False if the editor failed, ideally does not
             return if ``execv`` succeeds, and ``True`` if the
-            ``_exec_func`` does return successfully.
+            ``exec`` does return successfully.
         """
+        # gvim runs in the background by default so we force it to run
+        # in the foreground to ensure it gets attention.
+        if "gvim" in exe:
+            exe, *rest = args
+            args = [exe, "-f"] + rest
```

Comment:
Will this work if someone sets `EDITOR="gvim -f"`?

Reply:
Maybe but yeah it should add the art only once. Are you the gvim user?

Reply:
Nope, never used it before, just want to make sure it still works.

Reply:
ok this is fixed; test added as well.